mopidy_party/static/controller.js

```diff
@@ -153,49 +162,66 @@ angular.module('partyApp', [])
       }
       _index++;
     }
-
     $scope.$apply();
   };
 
   $scope.addTrackResult = function(track){
-
     $scope.tracks.push(track);
-    mopidy.tracklist.filter({'uri': [track.uri]}).done(
+	mopidy.tracklist.filter([{'uri': [track.uri]}]).done(
       function(matches){
         if (matches.length) {
           for (var i = 0; i < $scope.tracks.length; i++)
           {
             if ($scope.tracks[i].uri == matches[0].track.uri)
               $scope.tracks[i].disabled = true;
           }
-          $scope.$apply();
         }
       });
+    $scope.$apply();
   };
 
   $scope.addTrack = function(track){
-
     track.disabled = true;
 
-    var xmlHttp = new XMLHttpRequest();
-    xmlHttp.open( "POST", "/party/add", false ); // false for synchronous request
-    xmlHttp.send(track.uri);
-    var msgtype = 'success'
-    if (xmlHttp.status >= 400) {
-      track.disabled = false;
-      $scope.message = ['error', xmlHttp.responseText];
-    } else {
-      $scope.message = ['success', 'Next track: ' + track.name];
-    }
-    $scope.$apply();
+	$http.post('/party/add', track.uri).then(
+		function success(response){
+			$timeout(function() {
+				if (response.status < 400) {
```

Comment:
I am not sure I understand in which cases we would have a success with response.status >= 400?

https://docs.angularjs.org/api/ng/service/$http#$http-returns

Reply:
You're right. :) Unnecessary.

Reply:
This is a good example of why reviews exist ;)

Both your comments will imply a small change to the PR. I can do this tonight. If you would rather do it yourself, it is also fine :)